src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py

```diff
@@ -2150,12 +2174,15 @@ def _get_network_plugin(self, enable_validation: bool = False) -> Union[str, Non
             ) = self._get_pod_cidr_and_service_cidr_and_dns_service_ip_and_docker_bridge_address_and_network_policy(
                 enable_validation=False
             )
+            network_plugin_mode = self._get_network_plugin_mode(enable_validation=False)
             if network_plugin:
-                if network_plugin == "azure" and pod_cidr:
+                if network_plugin == "azure" and pod_cidr and network_plugin_mode != "overlay":
                     logger.warning(
-                        'When using `--network-plugin azure` without the preview feature '
-                        '`--network-plugin-mode overlay` the provided pod CIDR "%s" will be '
-                        'overwritten with the subnet CIDR.', pod_cidr
+                        'The provided pod CIDR "%s" will be overwritten with the node subnet CIDR. '
+                        'To use a pod CIDR, please specify network plugin mode `overlay` or '
+                        'use network plugin `kubenet`. For more information about Azure CNI '
+                        'Overlay please see https://aka.ms/aks/azure-cni-overlay. This warning '
+                        'will become an error in the Build sprint release.', pod_cidr
```

Comment:
It will become an error in the Build sprint release, right?
If so, please provide me with a description of this breaking change, and we will include it in document [upcoming-breaking-changes](https://learn.microsoft.com/en-us/cli/azure/upcoming-breaking-changes) before the breaking change release

Reply:
it will if we complete this pr https://github.com/Azure/azure-cli/pull/26237